pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -16,76 +15,109 @@ func Validate(client *Client, ic *types.InstallConfig) error {
+		if ic.Platform.AlibabaCloud != nil {
+			mpool.SetWithFieldPath(ic.Platform.AlibabaCloud.DefaultMachinePlatform, field.NewPath("platform", "alibabacloud", "defaultMachinePlatform"))
+		}
+
 		if compute.Platform.AlibabaCloud != nil {
-			allErrs = append(allErrs, validateMachinePool(client, ic, fldPath.Child("platform", "alibabacloud"), compute.Platform.AlibabaCloud, compute.Replicas)...)
+			mpool.SetWithFieldPath(compute.Platform.AlibabaCloud, fldPath.Child("platform", "alibabacloud"))
 		}
+		allErrs = append(allErrs, validateMachinePool(client, ic, &mpool)...)
 	}
-
-	return allErrs.ToAggregate()
+	return allErrs
 }
 
-func validatePlatform(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+func validateMachinePool(client *Client, ic *types.InstallConfig, pool *mergedMachinePool) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if ic.AlibabaCloud.ResourceGroupID != "" {
-		allErrs = append(allErrs, validateResourceGroup(client, ic, path)...)
-	}
+	if len(pool.Zones) > 0 {
+		availableZones := map[string]bool{}
+
+		response, err := client.DescribeAvailableResource("Zone")
+		if err != nil {
+			return append(allErrs, field.InternalError(pool.zonesFieldPath, err))
+		}
+		for _, availableZone := range response.AvailableZones.AvailableZone {
+			if availableZone.Status == "Available" {
+				availableZones[availableZone.ZoneId] = true
+			}
+		}
 
-	if ic.Platform.AlibabaCloud.DefaultMachinePlatform != nil {
-		allErrs = append(allErrs, validateMachinePool(client, ic, path.Child("defaultMachinePlatform"), ic.Platform.AlibabaCloud.DefaultMachinePlatform, nil)...)
+		for idx, zone := range pool.Zones {
+			if !availableZones[zone] {
+				allErrs = append(allErrs, field.Invalid(pool.zonesFieldPath.Index(idx), zone, fmt.Sprintf("zone ID is unavailable in region %q", ic.Platform.AlibabaCloud.Region)))
+			}
+		}
+	}
+	// InstanceType and zones are related.
+	// If the availability zone is not available, the instanceType will not be validated.
+	if len(allErrs) == 0 {
+		allErrs = append(allErrs, validateInstanceType(client, pool.Zones, pool.InstanceType, pool.instanceTypeFieldPath)...)
 	}
 
 	return allErrs
 }
 
-func validateMachinePool(client *Client, ic *types.InstallConfig, fldPath *field.Path, pool *alibabacloudtypes.MachinePool, replicas *int64) field.ErrorList {
+func validateInstanceType(client *Client, zones []string, instanceType string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	var zones []string
-	response, err := client.DescribeAvailableResource("Zone")
+	zoneWithStock := map[string]bool{}
+
+	response, err := client.DescribeAvailableZoneByInstanceType(instanceType)
 	if err != nil {
 		return append(allErrs, field.InternalError(fldPath, err))
 	}
-	for _, zone := range response.AvailableZones.AvailableZone {
-		if zone.Status == "Available" {
-			zones = append(zones, zone.ZoneId)
-		}
+	if len(response.AvailableZones.AvailableZone) == 0 {
```

Comment:
This check should only be done if the user did not specify specific zones to use.
```suggestion
	if len(zones) == 0 && len(response.AvailableZones.AvailableZone) == 0 {
```

Reply:
I am sorry to realize this problem after the merge, I think it has nothing to do with whether the user specifies zones, this is just to judge whether the available zone of `instanceType` (default or user specified) exists
I have create a PR #5479 to fix it.